nautobot/docs/development/apps/api/platform-features/uniquely-identify-objects.md

```diff
@@ -0,0 +1,57 @@
+# Uniquely Identifying a Nautobot Object
+
+In Nautobot v1.X, the slug field of many models could be used to uniquely identify a specific instance in the database. This was a convenient way to reference an object, but the slug field had some drawbacks and slugs were ultimately removed in Nautobot v2.0. As a result, new patterns will have to be adopted when trying to retrieve specific objects from the database.
+
+## Primary Keys vs. Natural Keys
+
+When designing an application for Nautobot, one of the key decisions is whether to use a primary key or a natural key to uniquely identify individual model instances. Here are some trade-offs to consider:
+
+### Primary Keys
+
+The primary key is a concrete database field and is the primary method for referencing an object or relating multiple objects in the Nautobot database. Here are some advantages of using primary keys:
+
+- **Uniqueness:** Primary keys are guaranteed to be unique.
+- **Performance:** Primary keys are always indexed in Nautobot, which makes lookups very fast.
+
+However, primary keys in Nautobot use UUIDs and can be difficult to work with directly.
+
+### Natural Keys
+
+A natural key is an identifier that is based on the natural attributes of a record, such as a platforms's name or in the case of some Nautobot models, a combination of fields. For example, the Prefix model's natural key is formed using a combination of the `prefix` field and the associated Namespace's `name`. Here are some advantages of using natural keys:
+
+- **Usability:** Natural keys are more user-friendly and easier to remember.
+- **Portability:** Natural keys can be used to identify an object in multiple contexts, such as in external applications or in different databases.
+
+However, there are also some disadvantages to using natural keys:
+
+- **Complexity:** Using natural keys can make it more difficult to change the underlying data structure without breaking existing integrations.
+- **Performance:** Natural keys are not guaranteed to be indexed which can make lookups slower. Also, natural keys that use a combination of fields will require database joins to retrieve the object from the database.
+
+In general, the decision of whether to use a primary key or a natural key depends on the specific requirements of the application. If uniqueness is critical, or if performance is a concern, a primary key may be the best choice. If human-readability is important, or if portability is a concern, a natural key may be the best choice.
+
+### Using Primary Keys
+
+In Nautobot v2.0, all object view URLs use the primary key of the object. In the web UI and REST API, the object can be accessed by building a URL using the primary key. For example, to retrieve the device with primary key `00000000-0000-0000-0000-000000000000`, the REST API URL would be `/api/dcim/devices/00000000-0000-0000-0000-000000000000/`.
```

Comment:
Nit: Should we use an actual uuid here instead of a string of zeroes?

Reply:
This is a valid uuid and I feel that a random string of characters is more distracting than all zeroes